File: NetDeck/DetailViewManager.m
//
//  DetailViewManager.m
//  Net Deck
//
//  Created by Gereon Steffens on 15.03.13.
//  Copyright (c) 2014 Gereon Steffens. All rights reserved.
//

#import "DetailViewManager.h"

@interface DetailViewManager ()

@end

@implementation DetailViewManager

// -------------------------------------------------------------------------------
//	setDetailViewController:
//  Custom implementation of the setter for the detailViewController property.
// -------------------------------------------------------------------------------
- (void)setDetailViewController:(UIViewController<SubstitutableDetailViewController> *)detailViewController
{
    _detailViewController = detailViewController;
    
    // Update the split view controller's view controllers array.
    // This causes the new detail view controller to be displayed.
    UIViewController *navigationViewController = [self.splitViewController.viewControllers objectAtIndex:0];
    NSArray *viewControllers = [[NSArray alloc] initWithObjects:navigationViewController, _detailViewController, nil];
    self.splitViewController.viewControllers = viewControllers;
}

#pragma mark -
#pragma mark UISplitViewDelegate

// -------------------------------------------------------------------------------
//	splitViewController:shouldHideViewController:inOrientation:
// -------------------------------------------------------------------------------
- (BOOL)splitViewController:(UISplitViewController *)svc 
   shouldHideViewController:(UIViewController *)vc 
              inOrientation:(UIInterfaceOrientation)orientation
{
    return UIInterfaceOrientationIsPortrait(orientation);
}

@end

@implementation SubstitutableNavigationController

@end

File: NetDeck/DetailViewManager.swift
//
//  DetailViewManager.swift
//  NetDeck
//
//  Created by Gereon Steffens on 26.05.16.
//  Copyright © 2017 Gereon Steffens. All rights reserved.
//

import UIKit

class DetailViewManager: NSObject, UISplitViewControllerDelegate {

    // Things for IB
    // The split view this class will be managing.
    @IBOutlet var splitViewController: UISplitViewController?

    // The presently displayed detail view controller.  This is modified by the various
    // view controllers in the navigation pane of the split view controller.
    @IBOutlet var detailViewController: UINavigationController? {
        willSet {
            self.detailViewController = newValue
            
            let navigationViewController = self.splitViewController?.viewControllers.first
            var viewControllers = [UIViewController]()
            if navigationViewController != nil {
                viewControllers.append(navigationViewController!)
            }
            if self.detailViewController != nil {
                viewControllers.append(self.detailViewController!)
            }
            self.splitViewController?.viewControllers = viewControllers
        }
    }
    
    func splitViewController(_ svc: UISplitViewController, shouldHide vc: UIViewController, in orientation: UIInterfaceOrientation) -> Bool {
        return UIInterfaceOrientationIsPortrait(orientation)
    }
}
